program/rust/src/tests/test_upd_permissions.rs

```diff
@@ -0,0 +1,89 @@
+use bytemuck::bytes_of;
+use solana_sdk::signer::Signer;
+
+use crate::c_oracle_header::{
+    CPubkey,
+    PermissionAccount,
+};
+use crate::instruction::{
+    OracleCommand,
+    UpdPermissionsArgs,
+};
+use crate::tests::pyth_simulator::PythSimulator;
+use crate::utils::pubkey_equal;
+
+#[tokio::test]
+async fn test_upd_permissions() {
+    let mut sim = PythSimulator::new_upgradable().await;
+
+    //Check that the program got deployed properly and can be interacted with
+    let mapping_keypair = sim.init_mapping().await.unwrap();
+    let product = sim.add_product(&mapping_keypair).await.unwrap();
+    let price = sim.add_price(&product, -8).await.unwrap();
+    assert!(sim.get_account(price.pubkey()).await.is_some());
+
+    let mut master_authority = CPubkey::new_unique();
+    let mut data_curation_authority = CPubkey::new_unique();
+    let mut security_authority = CPubkey::new_unique();
+
```

Comment:
I suggest adding some permissions tests here as well to ensure e.g., that only the upgrade authority can invoke this instruction.

I don't know the best way to set up those tests, but it's worth thinking about because we'll definitely need the same tests for all of the other instructions going forward.

Reply:
Made some helpers for changing the payer

Reply:
And added a test